src/usbsettings.h

```diff
@@ -40,29 +41,36 @@ class USBSettings: public QObject
 
     Q_ENUMS(Mode)
 
-    Q_PROPERTY(Mode currentMode READ currentMode NOTIFY currentModeChanged)
+    Q_PROPERTY(Mode currentMode READ currentMode WRITE setCurrentMode NOTIFY currentModeChanged)
     Q_PROPERTY(Mode defaultMode READ defaultMode WRITE setDefaultMode NOTIFY defaultModeChanged)
     Q_PROPERTY(QList<int> supportedUSBModes READ supportedUSBModes NOTIFY supportedUSBModesChanged)
 
 public:
+    /**
+     * Keep this in sync with usb_moded-dbus.h (for states) and usb_moded-modes.h (for modes),
+     * existing enum values taken from legacy qmsystem2 enum mapping for compatibility
+     **/
     enum Mode {
-        Connected = MeeGo::QmUSBMode::Connected,
-        DataInUse = MeeGo::QmUSBMode::DataInUse,
-        Disconnected = MeeGo::QmUSBMode::Disconnected,
-        MassStorage = MeeGo::QmUSBMode::MassStorage,
-        ChargingOnly = MeeGo::QmUSBMode::ChargingOnly,
-        OviSuite = MeeGo::QmUSBMode::OviSuite,
-        ModeRequest = MeeGo::QmUSBMode::ModeRequest,
-        Ask = MeeGo::QmUSBMode::Ask,
-        Undefined = MeeGo::QmUSBMode::Undefined,
-        SDK = MeeGo::QmUSBMode::SDK,
-        Developer = MeeGo::QmUSBMode::Developer,
-        MTP = MeeGo::QmUSBMode::MTP,
-        Adb = MeeGo::QmUSBMode::Adb,
-        Diag = MeeGo::QmUSBMode::Diag,
-        Host = MeeGo::QmUSBMode::Host,
-        Charger = MeeGo::QmUSBMode::Charger,
-        ConnectionSharing = MeeGo::QmUSBMode::ConnectionSharing
+        // States (from usb_moded-dbus.h)
+        Connected = 0,
```

Comment:
I've now extracted the old enum values from qmsystem2 headers, so that situations where it prints "Mode 3" in the UI shouldn't happen (as found by @pgerdt). This is the code I used to extract the numeric values for the enums (I then grouped the list by states and modes and removed the obsolete/unused ones):

```
#include "qmusbmode.h"

int main(int argc, char *argv[])
{
    struct {
        const char *name;
        int value;
    } m[] = {
#define SYM(x, y) { #x, y }
        SYM(Connected, MeeGo::QmUSBMode::Connected),
        SYM(DataInUse, MeeGo::QmUSBMode::DataInUse),
        SYM(Disconnected, MeeGo::QmUSBMode::Disconnected),
        SYM(MassStorage, MeeGo::QmUSBMode::MassStorage),
        SYM(ChargingOnly, MeeGo::QmUSBMode::ChargingOnly),
        SYM(OviSuite, MeeGo::QmUSBMode::OviSuite),
        SYM(ModeRequest, MeeGo::QmUSBMode::ModeRequest),
        SYM(Ask, MeeGo::QmUSBMode::Ask),
        SYM(Undefined, MeeGo::QmUSBMode::Undefined),
        SYM(SDK, MeeGo::QmUSBMode::SDK),
        SYM(Developer, MeeGo::QmUSBMode::Developer),
        SYM(MTP, MeeGo::QmUSBMode::MTP),
        SYM(Adb, MeeGo::QmUSBMode::Adb),
        SYM(Diag, MeeGo::QmUSBMode::Diag),
        SYM(Host, MeeGo::QmUSBMode::Host),
        SYM(Charger, MeeGo::QmUSBMode::Charger),
        SYM(ConnectionSharing, MeeGo::QmUSBMode::ConnectionSharing),
#undef SYM
    };

    for (auto &e: m) {
        printf("%s = %d,\n", e.name, e.value);
    }

    return 0;
}
```